1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.


In [11]:
import numpy as np
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
print(zp)
print(ks)


[ 35  45 190 200  40  70  54 150 120 110]
[401 574 874 919 459 739 653 902 746 832]


In [12]:
cov = np.mean(zp*ks) - np.mean(zp) * np.mean(ks)
print("Ковариация:", cov)

Ковариация: 9157.839999999997


In [16]:
print("Ковариация с помощью функции cov: ") 
np.cov (zp, ks, ddof = 0)

Ковариация с помощью функции cov: 


array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [17]:
pearson = cov / (np.std(zp, ddof = 0)*np.std(ks, ddof = 0))
print("Коэффициент корреляции Пирсона:", pearson)   

Коэффициент корреляции Пирсона: 0.8874900920739158


In [20]:
print("Коэффициент корреляции Пирсона с помощью функции corrcoef: ") 
np.corrcoef(zp, ks)

Коэффициент корреляции Пирсона с помощью функции corrcoef: 


array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [26]:
import pandas as pd

print("Коэффициент корреляции Пирсона с помощью pandas: ") 

zp_pd = pd.Series([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks_pd = pd.Series([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
zp_pd.corr(ks_pd)

Коэффициент корреляции Пирсона с помощью pandas: 


0.8874900920739162

In [27]:
ks_pd.corr(zp_pd)

0.8874900920739162

2. Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.



In [37]:
stud_iq = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
n = len(stud_iq)
n

10

In [39]:
a = 1 - 0.95
a

0.050000000000000044

In [40]:
v = n - 1
v

9

In [45]:
t = 2.262

In [46]:
mu = np.mean(stud_iq)
mu

118.1

In [47]:
std_stud_iq = np.std(stud_iq, ddof = 1)
std_stud_iq

10.54566788359614

In [48]:
mu + t*std_stud_iq/n**0.5

125.64339223691834

In [49]:
mu - t*std_stud_iq/n**0.5

110.55660776308164

Доверительный интервал [110.557, 125.643]

3. Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [51]:
D_G = 25
std_G = 25**0.5
n = 27
std_h = std_G/n**0.5
mu = 174.2
a =  1 - 0.95
Z = 1.96

In [52]:
mu + Z * std_h

176.08601087935276

In [53]:
mu - Z * std_h

172.31398912064722

Доверительный интервал [172.314, 176.086]